### PR TITLE
runtime: remove unimplemented CoCo configurations

### PR DESCRIPTION
**Pull request by fitzthum**
At the very beginning of the CoCo project, we added some speculative options to the config to guide development. It turns out that these don't match the features we implemented and that we will soon be porting to main. Let's remove them.

Technically the service_offload section still doesn't do much in main, but it will soon.

### CODE REVIEW

**Comment by fidencio** [comment]
> Technically the service_offload section still doesn't do much in main, but it will soon.

This is an interesting one, @fitzthum.  TIL that `service_offload` is irrelevant when using the nydus snapshotter, as offloading the image to the guest or to the guest is a snapshotter config, meaning that we most likely can drop it as well.

**Comment by fidencio** [comment]
IOW, I think we can drop the whole `image` section.

**Comment by fitzthum** [comment]
@ChengyuZhu6 Can you confirm that we won't need this `service_offload` flag?

**Comment by ChengyuZhu6** [comment]
> @ChengyuZhu6 Can you confirm that we won't need this `service_offload` flag?

@fitzthum , I'm very sorry for the delay in replying to you. I had tested it and I'm sure that we won't need `service_offload`.

**Comment by ChengyuZhu6** [comment]
@fitzthum One query here, but otherwise it looks good to me.

**Comment by ChengyuZhu6** [comment]
Same question like @fidencio. Is `Image` section no longer necessary? Should we maintain the flags other than `service_offload `?

**Review by stevenhorsman — APPROVED** [review]
Sorry I've missed/ignored this for a long time. I think we've discussed it on other issues that probably lead to this PR being created and I agree that the image configuration is not necessary and in fact confusing now, so the rebased version of this PR LGTM!

**Comment by fitzthum** [comment]
Ah yeah, I forgot about this two. Just gave it a rebase.

**Review by ChengyuZhu6 — APPROVED** [review]
LGTM thanks @fitzthum

File: src/runtime/pkg/katautils/config.go [hunk 1]
```diff
@@ -70,8 +70,7 @@ type tomlConfig struct {
 }
 
 type image struct {
-	Provision      string `toml:"provision"`
-	ServiceOffload bool   `toml:"service_offload"`
+	Provision string `toml:"provision"`
 }
```
Comment (stevenhorsman) [inline src/runtime/pkg/katautils/config.go:74]: I think we can get rid of the whole `image` struct now we've removed that block from the configuration.toml files?

Comment (fitzthum) [reply src/runtime/pkg/katautils/config.go]: Good point. Removed.

**Comment by stevenhorsman** [comment]
Sorry Tobin, it looks like the image timeout config has conflicted with these. If you are able to rebase today that would be great, otherwise let me know and I can try and take a look tomorrow

**Pull request merged**